BaragonUI/app/components/groupDetail/GroupDetail.jsx

```diff
@@ -2,35 +2,318 @@ import React, { Component, PropTypes } from 'react';
+            lable="Last Seen"
+            id="lastSeen"
+            key="lastSeen"
+            cellData={
+              // TODO timestamp from now
+              (agent) => (agent.lastSeenAt)
+            }
+            sortable={true}
+            />
+          {removeColumn()}
+        </UITable>
+      </div>
+    );
+  }
+
+  basePaths(basePaths, defaultDomain, groupName, editable) {
+    const removeButton = (basePath) => {
+      if (editable) {
+        return (
+          <RemoveBasePathButton
+            groupName={groupName}
+            basePath={basePath}
+          />
+        );
+      } else {
+        return null;
+      }
+    }
+    const pathToLink = (path) => {
+      if (defaultDomain) {
+        return <a target="_blank" href={`http://${defaultDomain}${path}`}>{path}</a>;
+      } else {
+        return <span>{path}</span>;
+      }
+    }
+    const renderBasePath = (element) => {
+      return (
+        <li className="list-group-item">
+          {removeButton(element)}
+          {pathToLink(element)}
+        </li>
+      );
+    }
+
+    const paths = this.asGroups(basePaths, renderBasePath);
+
+    return (
+      <div className="col-md-12">
+        <h4>Base Paths</h4>
+        {paths}
+      </div>
+    );
+  }
+
+  // Map a flat array into an array of arrays of size `size`, where the last
+  // array is as long as possible
+  // [1, 2, 3, 4, 5, 6, 7], 2 -> [[1, 2], [3, 4], [5, 6], [7]]
+  // Precondition: size != 0
+  chunk(arr, size) {
+    return _.chain(arr)
+      .groupBy((elem, index) => Math.floor(index / size))
+      .toArray()
+      .value();
+  }
+
+  asGroups(arr, itemRenderer) {
+    const rowRenderer = (row) => {
+      return (
+        <div className="col-md-3">
+          <ul className="list-group">
+            { row.map( itemRenderer ) }
+          </ul>
+        </div>
+      );
+    }
+
+    return this.chunk(arr, arr.length / 4).map(rowRenderer);
+  }
 }
 
 export default connect((state, ownProps) => ({
-  group: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data']),
+  domain: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'domain']),
+  domainsServed: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'domains']),
+  group: ownProps.params.groupId,
   basePaths: Utils.maybe(state, ['api', 'basePaths', ownProps.params.groupId, 'data']),
   targetCount: Utils.maybe(state, ['api', 'targetCount', ownProps.params.groupId, 'data']),
   agents: Utils.maybe(state, ['api', 'agents', ownProps.params.groupId, 'data']),
   knownAgents: Utils.maybe(state, ['api', 'knownAgents', ownProps.params.groupId, 'data']),
-  trafficSources: Utils.maybe(state, ['api', 'trafficSources', ownProps.params.groupId, 'data'])
-}))(rootComponent(GroupDetail, (props) => refresh(props.params.groupId)));
\ No newline at end of file
+  trafficSources: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'trafficSources']),
+  editable: true // TODO,
```

Comment:
we can actually grab this from `window.config` which is set up in our index.html. That is globally available to us to check.

I still need to implement the edit/no-edit piece of things, but a simple `if (window.config.allowEdit)` check should suffice when needed